src/main/java/eventprocessing/fileservices/CSVFileService.java

```diff
@@ -0,0 +1,78 @@
+package eventprocessing.fileservices;
+
+import eventprocessing.analysis.Analyser;
+import eventprocessing.models.Bucket;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class CSVFileService {
+    private final String outputCsvFile;
+    private final Analyser analyser;
+    private int ESTIMATED_LINE_LENGTH = 70;
+
+    public CSVFileService(String outputCsvFile) {
+        this.outputCsvFile = outputCsvFile;
+        this.analyser = new Analyser();
+    }
+
+    public void writeBucketDataToFile(Bucket bucketToWriteToFile) throws IOException {
```

Comment:
Not quite sure what you mean here - we use both, one when parsing the list of buckets at the start of the program, and one which runs regularly once the program has been running for 5 minutes.  Also both are called in the test (although it is probably not that clear what it would look like if either/both fail...